src/OrleansTestingHost/TestClusterOptions.cs

```diff
@@ -39,20 +49,57 @@ public TestClusterOptions(short initialSilosCount)
             this.BaseGatewayPort = ThreadSafeRandom.Next(40000, 50000);
         }
 
+        public static ConfigurationBuilder DefaultConfigurationBuilder()
+        {
+            var builder = new ConfigurationBuilder();
+            builder.AddInMemoryCollection(new Dictionary<string, string>
+            {
+                { nameof(ExtendedConfigurationOptions.DataConnectionString), "UseDevelopmentStorage=true" },
+                { nameof(ExtendedConfigurationOptions.LogsFolder), "logs" },
+                { nameof(ExtendedConfigurationOptions.TraceToConsole), "true" },
+                { nameof(ExtendedConfigurationOptions.InitialSilosCount), "2" }
+            });
+            return builder;
+        }
+
         /// <summary>
         /// The default initial silos count. See <see cref="TestClusterOptions"/>.
         /// </summary>
-        public static short DefaultInitialSilosCount { get; set; } = 2;
+        public static short GetInitialSilosCount(IConfiguration configuration)
+        {
+                short defaultValue;
+                return short.TryParse(configuration[nameof(ExtendedConfigurationOptions.InitialSilosCount)], out defaultValue)
```

Comment:
This is of course supported by the config lib, but I think a more maintainable means of doing this is to use ConfigurationBuilder.Bind to bind configured values directly to instance properties.